jsonschema-core.xml

```diff
@@ -2095,6 +2101,442 @@
+                </t>
+            </section>
+
+            <section title="Output Levels">
+                <t>
+                    This specification defines four levels of output.  See the *Output Structure*
+                    section for the requirements of each level.
+                    <list>
+                        <t>
+                            Flag - A boolean which simply indicates whether the schema was
+                            validated.
+                        </t>
+                        <t>
+                            Basic - Provides validation information in a flat list structure.
+                        </t>
+                        <t>
+                            Detailed - Provides validation information in a condensed hierarchical
+                            structure based on the structure of the schema.
+                        </t>
+                        <t>
+                            Verbose - Provides validation information in an uncondensed hierarchical
+                            structure that matches the exact structure of the schema.
+                        </t>
+                    </list>
+                    Implementations SHOULD specify which levels of validation they support.
+                </t>
+
+            </section>
+
+            <section title="Minimum Information">
+                <t>
+                    Beyond the simplistic "Flag" output, additional information is useful to aid in
+                    debugging a schema or instance.  Each sub-result SHOULD contain the information
+                    contained within this section at a minimum.
+                </t>
+                <t>
+                    A single object that contains all of these components is considered an
+                    output unit.
+                </t>
+                <t>
+                    Implementations MAY elect to provide additional information.
+                </t>
+
+                <section title="Keyword Relative Location">
+                    <t>
+                        The relative location of the validating keyword that follows the validation
+                        path.  The value MUST be expressed as a JSON Pointer, and it MUST include
+                        "$ref" or "$recursiveRef" segments for any reference jumps.
+                    </t>
+                    <figure>
+                        <artwork>
+<![CDATA[
+#/properties/minLength/$ref/minimum
+]]>
+                        </artwork>
+                    </figure>
+                    <t>
+                        The JSON key for this information is "keywordLocation".
+                    </t>
+                </section>
+
+                <section title="Keyword Absolute Location">
+                    <t>
+                        The absolute, dereferenced location of the validating keyword.  The value MUST
+                        be expressed as an absolute URI, and it MUST NOT include "$ref" or
```

Comment:
These are more generically termed "by-reference applicators" per section 4.3.5 Referenced and Referencing Schemas.  So maybe

> MUST NOT include by-reference applicators such as "$ref" or "$recursiveRef"

to keep it general?

Reply:
✔️